src/components/header/header.js

```diff
@@ -1,16 +1,29 @@
+import { useAppContext } from "../../context/AppContext";
+import { Sun, Moon } from "react-feather";
+import { useEffect, useState } from "react";
+
 export default function Navbar() {
+  const { darkMode, toggleTheme } = useAppContext();
+  let [dark, setDark] = useState("light");
+  useEffect(() => {
+    setDark(darkMode);
+  }, [darkMode]);
```

Comment:
Again why do you need a separate state for the theme when you already have context.